python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -307,12 +307,23 @@ def run(self, splash, version, **kwargs):
         self.startup_descriptor = kwargs.get("startup_descriptor")
         server = kwargs.get("server")
 
-        # Log usage statistics about the Shotgun Desktop executable and the desktop startup.
-        sgtk.util.log_user_attribute_metric("tk-framework-desktopstartup", self.startup_version)
-        sgtk.util.log_user_attribute_metric("Shotgun Desktop version", self.app_version)
-        # If a server is passed down from the desktop startup, it means we won't be using the engine-based
-        # websocket server.
-        sgtk.util.log_user_attribute_metric("Engine-Websockets", "no" if server else "yes")
+        try:
+            # Log usage statistics about the Shotgun Desktop executable and the desktop startup.
+            #
+            # First we update `host_info` property so subsequent metrics can benefit
+            # having the updated information. A special case is made for for Desktop
+            # as we do want both versiond but don't want to create another metric field.
+            # We are then combining both versions into single version string.
+            self._engine._host_info.update({
```

Comment:
why `update`? Should it just set the value? `self._engine._host_info["version"] = ...`

Reply:
We don't have the version at that time, it comes later on in the site implementation 'run' method.

Reply:
Update is typically used when you have two dictionaries and you want to update one with the contents of the other. In this case, you have the host info dictionary and a value you need to set for one of its keys. In this case, `update` works but I think setting the value is better. 

```python
version = "%s / %s" % (self.app_version, self.startup_version)
self._engine._host_info["version"] = version
```